### DISCUSSION TITLE
REQ: Make it easier to step to a certain player to spectate.

### DISCUSSION THREAD

**Discussion by Pulseczar1** [ezQuake/ezquake-source#497] [Ideas]
**The Problem**
It's always tedious to select a certain player for tracked spectating. This tedium increases as player count increases. Currently, you have to keep pressing jump until the game "randomly" selects the player you want. If you try to go fast and accidentally skip the desired player, you have to start over.

**Describe the solution you'd like**
I would like a non-random way of selecting a player to spectate. An easy thing to implement, I think, would be to have jump take you to the next player that currently follows the currently tracked player on your scoreboard.

**Describe alternatives you've considered**
There are other ways to do it that would work better, but my suggestion, I think, is very easy to implement, and it would suit me very well, and I imagine, others, too.

**Additional context**
Pressing jump at this point in time (see screenshot), should take you to the player, Data, rather than to a "random" player. This would make it easy to anticipate how many times you have to press jump to get the player you want.
![ezquake005](https://user-images.githubusercontent.com/49758921/101677913-3741dc80-3a2b-11eb-8920-721d8b74edf4.png)

**Impact on server or mods**
I don't think this has anything to do with server-side functionality. My guess is that the client is currently just stepping through players, either by client/edict number, or by user ID.

**Comment by hemostx** [comment]
An alternative to this is an option to enable the mouse when the scoreboard is visible ; you can then click on the player you wish to track. There's a way of doing this in CSGO with a modifier key, I think - I'd have to take another look but IIRC it worked well.

**Comment by Calinou** [comment]
> There's a way of doing this in CSGO with a modifier key, I think - I'd have to take another look but IIRC it worked well.

In CS:GO, it works by right-clicking while the scoreboard is displayed (but you can also change it to use another key).

That said, it sounds difficult to implement in ezQuake considering the mouse-driven menu code is most likely not reusable in the scoreboard.

**Comment by CecilaOlsen** [comment]
Ich verstehe den Frust absolut. Wenn man versucht, einem bestimmten Spieler zuzusehen, und das System die Spieler scheinbar willkürlich durchläuft, wird es schnell zum Ratespiel. Die Idee, die Spieler anhand der Anzeigetafel-Reihenfolge durchzugehen, ist sehr sinnvoll. So wäre es zumindest vorhersehbar, und man wüsste genau, wie oft man klicken muss, um den gewünschten Spieler zu erreichen.

Der Vorschlag, die Anzeigetafel mit der Maus zu bedienen, ist ebenfalls interessant. Direkt auf einen Namen zu klicken, wäre zwar die intuitivste Lösung, aber ich verstehe, dass die Umsetzung schwieriger sein könnte, wenn der UI-Code nicht dafür ausgelegt ist. Wunderwins Casino https://wunderwins-casino.win/ steht für zügigen Spielspaß und eine klare Benutzerführung. Selbst eine einfache, geordnete Spielerauswahl wäre schon ein großer Fortschritt in puncto Benutzerfreundlichkeit.

**Comment by JesseReed123** [comment]
Je comprends parfaitement la frustration. Actuellement, sélectionner un joueur à suivre en spectateur est aléatoire et devient très fastidieux lorsque le nombre de joueurs augmente. Une solution simple serait de faire en sorte que la touche "jump" vous fasse passer directement au joueur suivant sur le scoreboard, dans l’ordre où ils suivent le joueur actuellement suivi. Cela permettrait https://iledecasino.fr/ de savoir exactement combien de fois appuyer pour atteindre le joueur désiré, rendant le spectateur beaucoup plus pratique et intuitif, surtout pour les tournois ou streams.

**Comment by Pulseczar1** [comment]
I'm not sure how I didn't notice the casino link the first time I translated the German comment, but I noticed it this time after returning for what seems to be basically the same comment, but written in French this time. Since these are just basic summaries and agreements with the discussion above, I now believe these are AI comments, especially given that strange casino link in the middle of the German comment.
Edit: Oh, the casino link advertisement was edited into the comment 2 days after the comment was first posted.
